src/extension.ts

```diff
@@ -63,12 +86,33 @@ export function activate(context: ExtensionContext) {
     });
 
     workspace.onDidSaveTextDocument(doc => {
-        if (ameba && ameba.config.onSave) ameba.execute(doc);
+        if (ameba && ameba.config.onSave && validDidSaveDocument(doc)) {
+            outputChannel.appendLine(`[Save] Running ameba on ${getRelativePath(doc)}`)
+            ameba.execute(doc);
+        }
     });
 
     workspace.onDidCloseTextDocument(doc => {
         ameba && ameba.clear(doc);
     });
 }
 
-export function deactivate() {}
+export function deactivate() { }
+
+function getRelativePath(document: TextDocument): string {
+    const space: WorkspaceFolder =
+        workspace.getWorkspaceFolder(document.uri) ?? noWorkspaceFolder(document.uri)
+    return path.relative(space.uri.fsPath, document.uri.fsPath)
+}
+
+export function noWorkspaceFolder(uri: Uri): WorkspaceFolder {
+    return {
+        uri: Uri.parse(path.dirname(uri.fsPath)),
+        name: path.basename(path.dirname(uri.fsPath)),
+        index: -1
+    }
+}
+
+function validDidSaveDocument(doc: TextDocument): boolean {
```

Comment:
```suggestion
function isValidCrystalDocument(doc: TextDocument): boolean {
```

Reply:
Once on type is added, only the languageId needs to be checked, meaning this method name will either be misleading or changed back to what it is now

Reply:
Let's worry about it then.